ts/nni_manager/training_service/reusable/environments/dlcEnvironmentService.ts

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+'use strict';
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as component from '../../../common/component';
+import { getLogger, Logger } from '../../../common/log';
+import { ExperimentConfig, DlcConfig, flattenConfig } from '../../../common/experimentConfig';
+import { ExperimentStartupInfo } from '../../../common/experimentStartupInfo';
+import { DlcClient } from '../dlc/dlcClient';
+import { DlcEnvironmentInformation } from '../dlc/dlcConfig';
+import { EnvironmentInformation, EnvironmentService } from '../environment';
+import { EventEmitter } from "events";
+import { FileCommandChannel } from '../channels/fileCommandChannel';
+import { MountedStorageService } from '../storages/mountedStorageService';
+import { Scope } from 'typescript-ioc';
+import { StorageService } from '../storageService';
+
+interface FlattenDlcConfig extends ExperimentConfig, DlcConfig { }
+
+/**
+ * Collector DLC jobs info from DLC cluster, and update dlc job status locally
+ */
+@component.Singleton
+export class DlcEnvironmentService extends EnvironmentService {
+
+    private readonly log: Logger = getLogger('dlcEnvironmentService');
+    private experimentId: string;
+    private config: FlattenDlcConfig;
+
+    constructor(config: ExperimentConfig, info: ExperimentStartupInfo) {
+        super();
+        this.experimentId = info.experimentId;
+        this.config = flattenConfig(config, 'dlc');
+        component.Container.bind(StorageService).to(MountedStorageService).scope(Scope.Singleton);
+        const storageService = component.get<StorageService>(StorageService)
+        const remoteRoot = storageService.joinPath(this.config.localStorageMountPoint, 'nni-experiments', this.experimentId);
+        const localRoot = storageService.joinPath(this.config.localStorageMountPoint, 'nni-experiments');
+        storageService.initialize(localRoot, remoteRoot);
+    }
+
+    public get hasStorageService(): boolean {
+        return true;
+    }
+
+    public initCommandChannel(eventEmitter: EventEmitter): void {
+        this.commandChannel = new FileCommandChannel(eventEmitter);
+    }
+
+    public createEnvironmentInformation(envId: string, envName: string): EnvironmentInformation {
+        return new DlcEnvironmentInformation(envId, envName);
+    }
+
+    public get getName(): string {
+        return 'dlc';
+    }
+
+    public async refreshEnvironmentsStatus(environments: EnvironmentInformation[]): Promise<void> {
```

Comment:
I suppose the returned promise should be resolved when refreshing is done. @SparkSnail
If thats the case, this function should return `Promise.all(...)`.

Reply:
this is to keep same as refreshEnvironmentsStatus of KubernetesEnvironmentService/AMLEnvironmentService.